classes/event/logging_succeed.php

```diff
@@ -0,0 +1,65 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * This event gets triggered when log record added.
+ *
+ * @package    tool_etl
+ * @copyright  2019 Dmitrii Metelkin <dmitriim@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_etl\event;
+
+use core\event\base;
+use tool_etl\logger;
+
+defined('MOODLE_INTERNAL') || die();
+
+class logging_succeed extends base {
+
+    /**
+     * @inheritdoc
+     */
+    protected function init() {
+        $this->data['crud'] = 'u';
+        $this->data['edulevel'] = self::LEVEL_OTHER;
+        $this->data['objecttable'] = logger::TABLE;
+        $this->context = \context_system::instance();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function get_name() {
+        return get_string('eventlogrecordadded', 'tool_etl');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function get_description() {
+        $runid = $this->data['other']['runid'];
+        $taskid = $this->data['other']['taskid'];
+        $logtype = $this->data['other']['logtype'];
+        $element = $this->data['other']['element'];
+        $action = $this->data['other']['action'];
+        $info = $this->data['other']['info'];
+
```

Comment:
So I'm just thinking out loud if any of these 'other' fields can be moved (or duplicated) up into the explicit event fields, ie

https://docs.moodle.org/dev/Event_2#Properties

eg the core 'action' field could be the workflow action, maybe target could be $element. I guess the rough guide here is to look purely at the moodle log gui and see what comes out for a bunch of workflow actions and see if you can make sense of it without needing more context.

Also objecttable, and objectid could link to either the internal log table or the workflow records. I'm a little hazy on what core actually does with this.

Reply:
What's a point for that?